### PR TITLE
Make `OmegaConfigLoader` the default config loader

### LINKED ISSUE #2949: Make `OmegaConfigLoader` the default config loader
## Description
The `ConfigLoader` and `TemplatedConfigLoader` will be removed in https://github.com/kedro-org/kedro/issues/2692, so the new default should be `OmegaConfigLoader`. 

## To be decided
We could either do this in the 18 series before the `ConfigLoader` and `TemplatedConfigLoader` are removed or do it as part of `0.19.0`.

### PR DESCRIPTION
**Pull request by ankatiyar**
## Description
Resolve #2949 

## Development notes

- Updated settings
- Updated context
- Updated session
- Updated tests
- Updated template project

Will update docs after removal of `TemplatedConfigLoader` and `ConfigLoader` since it requires some rewriting. 

## Checklist

- [ ] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [ ] Signed off each commit with a [Developer Certificate of Origin (DCO)](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/managing-repository-settings/managing-the-commit-signoff-policy-for-your-repository)
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes
- [ ] Checked if this change will affect Kedro-Viz, and if so, communicated that with the Viz team

### CODE REVIEW

**Review by merelcht — APPROVED** [review]
🥳

File: kedro/framework/context/context.py [hunk 1]
```diff
@@ -166,13 +166,13 @@ class KedroContext:
 
     _package_name: str
     project_path: Path = field(converter=_expand_full_path)
-    config_loader: ConfigLoader
+    config_loader: OmegaConfigLoader
```
Comment (astrojuanlu) [inline kedro/framework/context/context.py:169]: So the interface is the OmegaConfigLoader and not the AbstractConfigLoader?

Comment (ankatiyar) [reply kedro/framework/context/context.py]: RTD was causing issues with AbstractConfigLoader but i'm trying to update it

Comment (noklam) [reply kedro/framework/context/context.py]: I believe this should me the abstract class - some user point this out a while ago but I can't find the issue.

**Review by noklam — APPROVED** [review]

**Pull request merged**